src/mdframe/reader.py

```diff
@@ -91,9 +106,9 @@ def main():
 
     # parse schema url to determine whether Url or Path passed
     parsed_schema = urlparse(args.schema)
-    if parsed_schema.scheme in ['http', 'https']:
-        schema = Url(parsed_schema)
-    else:
+
+    # only change schema if it is not a Path (if it is a Url, just pass it as a string)
+    if parsed_schema.scheme not in ['http', 'https']:
         try:
```

Comment:
Why is there a `try` block here? I'd put the URL/Path validator into a separate function that would return a `URL` or `Path` object based on which entity it is. This way you get reusable validation and you explicitly work with an object that is unambiguously `Path` or `URL`.

I'd avoid `try`/`except` like fire. It's always better to perform conditional checks with `if/elif/else` or `match` statements than relying on the error catching mechanism. It also makes the code a little clearer to read. Consider this particular example here. Because you're using a bare `try` block, you're not being explicit about what kind of error was expected from this statement. I can guess it's going to be an error related to `args.schema` being a URL and wrapping it into a `Path` would make things crash. But it's not immediately clear when you look at it.

Reply:
Yep, reformatted this setup so that it's now cleaner than before.

I think we still might have to modify the behavior of the try-except in the `load_metadata_files` method just to take care of the case where a file is invalid and allow the program to skip over it instead of having to crash because of an invalid file.

What I think will be a worthwile addition is to have the error message inside `load_metadata_files` to say which metadata file is causing the exception. That way, we can also concurrently rectify issues in our metadata files as we go.

I'll add that in as well. :rocket: